kernel/src/scan/mod.rs

```diff
@@ -107,11 +121,16 @@ pub struct ScanResult {
     pub mask: Option<Vec<bool>>,
 }
 
-/// Scan uses this to set up what kinds of columns it is scanning. A reference to the field is
-/// stored so we can get the name and data_type out when building the read expression
-pub enum ColumnType<'a> {
-    Selected(&'a StructField),
-    Partition(&'a StructField),
+/// Scan uses this to set up what kinds of columns it is scanning. For Selected and Partition, the
```

Comment:
note i've included both ways we could do things here. `Selected` stores an actual copy of the field which we have to `clone()` when we do `ScanBuilder::build()`, but does simplify things slightly later when we need the field. `Partition` just stores the index, and then when needed we can index into the schema to get the actual field. This introduces the problem that things could get out of sync (although that would likely indicate bigger issues), but should be more efficient that cloning the whole field.